### PR TITLE
feat: add workflow for uploading installers to release assets

### PR DESCRIPTION
**Pull request by AnqiPang**
Issue #, if available:

*Description of changes:*
This workflow is to be run after the `release-installer.yaml` workflow completes. And after this workflow completes, the next step is to run the `release-homebrew.yaml` workflow. All the three workflows should be merged together once this [issue](https://github.com/runfinch/finch/issues/106) is fixed.
*Testing done:*
Tested the workflow within my own fork.

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/release-installer.yaml [hunk 1]
```diff
@@ -106,4 +106,4 @@ jobs:
       - name: Silently uninstall
         run: echo 'y' | sudo bash /Applications/Finch/uninstall.sh
       - name: Delete installer
-        run: rm -rf Finch-${GITHUB_REF_NAME}-x86_64.pkg
+        run: rm -rf Finch-${GITHUB_REF_NAME}-x86_64.pkg
```
Comment (vsiravar) [inline .github/workflows/release-installer.yaml:109]: nit: Newline is deleted.

File: .github/workflows/upload-installer-to-release.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.ROLE }}
+          role-session-name: download-installer-session
+          aws-region: ${{ secrets.REGION }}
+      - name: Download installers
+        run: |
+          aws s3 cp s3://${{ secrets.INSTALLER_PRIVATE_BUCKET_NAME }}/Finch-${GITHUB_REF_NAME}-aarch64.pkg Finch-${GITHUB_REF_NAME}-aarch64.pkg
```
Comment (vsiravar) [inline .github/workflows/upload-installer-to-release.yaml:21]: What does `GITHUB_REF_NAME` evaluate to in this context? Is it the same as ${{ github.ref_name }} ?

File: .github/workflows/upload-installer-to-release.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          aws s3 cp s3://${{ secrets.INSTALLER_PRIVATE_BUCKET_NAME }}/Finch-${GITHUB_REF_NAME}-x86_64.pkg Finch-${GITHUB_REF_NAME}-x86_64.pkg
+      - name: Upload installers to release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: ${{  github.ref_name }}
+          files: |
+            Finch-${{  github.ref_name }}-aarch64.pkg
```
Comment (sam-berning) [inline .github/workflows/upload-installer-to-release.yaml:28]: Should we clean up the files after we upload them to the release?

Comment (AnqiPang) [reply .github/workflows/upload-installer-to-release.yaml:28]: Yes you are right. Thanks for pointing out.

File: .github/workflows/upload-installer-to-release.yaml (hunk 2 above)
Comment (AnqiPang) [reply .github/workflows/upload-installer-to-release.yaml:21]: It's the same. [GITHUB_REF_NAME](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) is a default environment variable, and [github.ref_name](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) is the github context accessible via env vars as well. Their value is the tag that triggered this workflow.

**Review by sam-berning — APPROVED** [review]

**Review by vsiravar — APPROVED** [review]

**Comment by ningziwen** [comment]
nit: should be this "ci" isntead of "feat"?

**Comment by AnqiPang** [comment]
I think we can fix the release notes at the release. WDYT? @ningziwen

**Comment by sam-berning** [comment]
You can change the commit message release-please uses with this process: https://github.com/googleapis/release-please#how-can-i-fix-release-notes

**Comment by AnqiPang** [comment]
Following this [example PR](https://github.com/runfinch/common-tests/pull/11), I updated the merged PRs #208 and #202 with overridden commit message. Need to validate the release note is fixed after the next release-please run.

**Pull request merged**